sdk/azcore/runtime/pager.go

```diff
@@ -88,3 +90,25 @@ func (p *Pager[T]) NextPage(ctx context.Context) (T, error) {
 func (p *Pager[T]) UnmarshalJSON(data []byte) error {
 	return json.Unmarshal(data, &p.current)
 }
+
+// FetcherHelper is a helper containing boilerplate code to simplify creating a PagingHandler[T].Fetcher.
+func FetcherHelper(ctx context.Context, pl Pipeline, nextLink string, createReq func(context.Context) (*policy.Request, error)) (*http.Response, error) {
```

Comment:
Mabye we should just name this `Fetcher`?  I'm not super thrilled by the suffix but can live with it.

Reply:
It's an unfortunate name but it does accurately describe this thing. I know you looked at this already, but I don't have a firm opinion on where to draw the line between generating code and consolidating it into azcore, so I still wonder whether it's better to generate this (once per package?) instead of adding it to the public API. If it's just the increase in generated LOC that's troubling, could we generate a generic `Fetcher` factory?

Reply:
> where to draw the line between generating code and consolidating it into azcore

This is a great point and probably something we should try to formalize.

The way I think about it is that `azcore` contains general-purpose building blocks and the generated code assembles them in a domain-specific way.  So, I would argue that if we have repetitive chunks of general-purpose code being generated, we should move those chunks to `azcore`.

@JeffreyRichter what are your thoughts on this?

Reply:
Back to naming, perhaps this should be `FetcherForNextLink` (or something like that) as it only works for paged operations that use the "next link" pattern.

Reply:
My immediate thought is that this will be in the public surface area for the runtime package, but no customer should ever use it directly, right?
So, the docs for it would say to not use it and there would be no example code for it either, right?
And, as always, if we code-gen it today, then we could move it here tomorrow, but we can never the reverse. So, I'm prefer the conservative approach.

Reply:
I renamed this to `FetcherForNextLink` in hopes that one day we'll also have a `FetcherForContinuationToken` if/when we support modeling this.

Jeff and I discussed when something should go in `azcore/runtime` vs generated code.  Here are the guiding principles we came up with.  Not really a prescriptive list, but things to keep in mind when weighing the decision.
- Generated code
  - Pros: immune to breaking changes, decentralized code insulates packages from regressions, no new public surface area
  - Cons: duplicates code, rolling out a bug-fix requires fixing/releasing the code generator, then regenerating the world
- Code in `azcore/runtime`
  - Pros: no duplicated code, bug fixes are centralized
  - Cons: regressions in shared code can break everybody, no breaking changes means potential for deprecated code

As an aside, the fact that `runtime` contains some APIs intended for use by customers was a mistake and will be cleaned up in https://github.com/Azure/azure-sdk-for-go/pull/21406.